processor/src/decorators/mod.rs

```diff
@@ -186,6 +188,60 @@ impl Process {
 
         Ok(())
     }
+
+    /// Given a polynomial ( in NTT form ), over quadratic extension field, of domain size
+    /// power of 2, on stack top, this routine interpolates the polynomial using inverse
+    /// Number Theoretic Transform, producing the polynomial in its coefficient form, of
+    /// provided domain size, which is written on the advice tape.
+    ///
+    /// Input stack state should look like
+    ///
+    /// `[result_poly_domain_size, input_poly_domain_size, input_poly_begin_mem_address, ...]`
+    ///
+    /// - `result_poly_domain_size` must be a power of 2.
+    /// - `input_poly_domain_size` must be a power of 2.
+    /// - `result_poly_domain_size <= input_poly_domain_size`
+    /// - Only starting memory address i.e. first code-word's address is provided on stack, consecutive
+    ///  memory addresses are expected to be holding remaining `input_poly_domain_size - 1` many code-words.
+    /// - Each memory address can hold two code-words i.e. each code-word consists of two elements ∈ Z_q | q = `2^64 - 2^32 + 1`
+    ///
+    /// Final advice tape should look like
+    ///
+    /// `[coeff_0, coeff_1, ..., coeff_{n-1}, ...]` | n = result_poly_domain_size
+    ///
+    /// Program which is requesting this non-deterministic computation should read `coeff0`
+    /// first i.e. `coeff{n-1}` should be seen at the very end.
+    fn inject_ext2_intt_result(&mut self) -> Result<(), ExecutionError> {
+        let out_poly_len = self.stack.get(0).as_int() as usize;
+        let in_poly_len = self.stack.get(1).as_int() as usize;
+        let in_poly_addr = self.stack.get(2).as_int();
+
+        if !in_poly_len.is_power_of_two() {
+            return Err(ExecutionError::NonPowerOf2PolyDomainSize(in_poly_len as u64));
+        }
+        if !out_poly_len.is_power_of_two() {
+            return Err(ExecutionError::NonPowerOf2PolyDomainSize(out_poly_len as u64));
+        }
+
+        let mut poly = Vec::with_capacity(in_poly_len);
+        for i in 0..(in_poly_len >> 1) {
+            let word = self
+                .get_memory_value(self.system.ctx(), in_poly_addr + i as u64)
+                .ok_or_else(|| ExecutionError::CallerNotInSyscall)?;
```

Comment:
I actually made an error in my previous comment here. The error returned should be `InaccessibleMemoryAddress`.

Reply:
Sorry, I also didn't notice that - fixed in f4e857e1